### PR TITLE
Update release strategy

### PR DESCRIPTION
**Pull request by jose-cortina**
**Description**

Changes proposed in this pull request:

- Update the release strategy document to match the current state of the project and teams
- Added some details about handling critical issues found on a release candidate and how we proceed with this

**Related issue(s)**

See also #400

### CODE REVIEW

File: guidelines/releases-guidelines/01-release-strategy.md [hunk 1]
```diff
@@ -114,9 +114,9 @@ At this time, the Release Manager publicly communicates the planned release sche
...
+After completing the planning, the Product Lead explicitly hands over the release to the engineering teams during the handover meeting. At this point, the development phase of the new release officially starts.
 
-During the development phase, the Release Manager keeps track of the development with the help of the Scrum of Scrums (SoS) meetings held twice a week. This allows the Release Manager to take early actions required to keep the release on track. When closer to the release date, the Release Manager, the Head of Capabilities, and the Lead Product Engineer discuss the status of the release and decide if there are any additional checkpoints or actions to address.
+During the development phase, the Release Manager keeps track of the development with the help of the Scrum of Scrums (SoS) meetings held twice a week. This allows the Release Manager to take early actions required to keep the release on track. When closer to the release date, the Release Manager and the Product Lead discuss the status of the release and decide if there are any additional checkpoints or actions to address.
```
Comment (Disper) [inline guidelines/releases-guidelines/01-release-strategy.md]: I'm not sure if this is worth mentioning as the Scrum of Scrum meetings are internal.

Comment (jose-cortina) [reply guidelines/releases-guidelines/01-release-strategy.md]: Updated the document

File: guidelines/releases-guidelines/01-release-strategy.md [hunk 2]
```diff
@@ -114,27 +114,27 @@ At this time, the Release Manager publicly communicates the planned release sche
...
 
 During the execution of a release, all automated end-to-end test scenarios are validated. If any scenario fails, the release process automatically stops and no release is possible.
 
-The Release Manager notifies the Kyma community on a public Slack channel that the release candidate is available, requesting the community to validate it and provide their short-term feedback. The deadline for the feedback depends on the release type. Ideally, the final release is the same as the last release candidate, avoiding any additional changes. If there are release-related issues identified, the Release Manager decides if they are blockers for the release.
+The Release Manager notifies the Kyma community on a public Slack channel when a release candidate is available, at which time the community can validate it and provide their feedback. If there are release-related issues identified, the Release Manager decides if any represent blockers for the release.
```
Comment (mmitoraj) [inline guidelines/releases-guidelines/01-release-strategy.md]:
```suggestion
The Release Manager notifies the Kyma community on the public **release** Slack channel when a release candidate is available, at which time the community can validate it and provide their feedback. If there are release-related issues identified, the Release Manager decides if any of them is a blocker for the release.
```

File: guidelines/releases-guidelines/01-release-strategy.md [hunk 3]
```diff
@@ -152,9 +152,7 @@ When planning the release scope, all persons involved in the release must take t
 
 ### Version maintenance
 
-There is no guaranteed support for the bug fixes in the previous Kyma versions. The default strategy is to upgrade to the latest version.
-
-The Kyma project does not plan to provide any patch releases before the 1.0 production release and encourages the community to always upgrade to the latest release.
+There is no guaranteed support for the bug fixes in the previous Kyma versions. The default strategy is to upgrade to the latest version. The Kyma project encourages the community to always upgrade to the latest release.
```
Comment (mmitoraj) [inline guidelines/releases-guidelines/01-release-strategy.md]:
```suggestion
There is no guaranteed support for the bug fixes in the previous Kyma versions. The default strategy is to upgrade to the latest version. The Kyma team encourages the community to always upgrade to the latest release.
```

File: guidelines/releases-guidelines/04-release-communication.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ At the beginning of the release process, the team responsible for the release pi
 
 ## Communication channels
 
-The Kyma team must be notified about the planned dates of both release candidates, as well as the date of the final release. Apart from that, the Release Master must inform about any occurrences regarding the release process. After the release is out, the Release Master sends notification about the new release on the external Kyma community [Slack workspace](http://slack.kyma-project.io/) **general** channel.
+The Kyma team must be notified about the planned dates of both release candidates, as well as the date of the final release. Apart from that, the Release Master must inform about any occurrences regarding the release process. After the release is out, the Release Master sends notification about the new release on the external Kyma community [Slack workspace](http://slack.kyma-project.io/) **release** channel.
```
Comment (mmitoraj) [inline guidelines/releases-guidelines/04-release-communication.md]:
```suggestion
The Release Manager must notify the Kyma team about the planned dates of creating both release candidates, as well as the date of the final release. Apart from that, the Release Master must inform about the progress of the release process. After the release is out, the Release Master sends notification about the new release on the external **release** [Slack](http://slack.kyma-project.io/) channel.
```
I think that now not only the information that the release is out but whole communication about the release takes place on the public **release** Slack channel, isn't it? Also, line 137 says that it is the Release Manager that notifies the Kyma community on the public **release** Slack channel when a release candidate is available. So maybe we could add the Release Manager to this section as well?

File: guidelines/releases-guidelines/01-release-strategy.md [hunk 5]
```diff
@@ -62,11 +60,11 @@ In Kyma, we follow the test strategy according to which you can verify all funct
 
 The Kyma community must provide the highest possible test coverage to minimize any potential risks and ensure the release is functional and stable.
 
-Whenever possible, consider implementing tests that go beyond verifying the functionality of the system. This includes integration, performance, and end-to-end tests.
+Consider implementing tests that go beyond verifying the functionality of the system. This includes integration, performance, and end-to-end tests.
 
 ### Compliance rules
 
-All Kyma organization members involved in the release process comply with contribution rules defined for Kyma and ensure that all checks built into the continuous integration process pass. These requirements represent the absolute minimal acceptance criteria needed for a release from a compliance perspective. These rules can change as Kyma approaches the 1.0 release.
+All Kyma organization members involved in the release process comply with contribution rules defined for Kyma and ensure that all checks built into the continuous integration process pass. These requirements represent the absolute minimal acceptance criteria needed for a release from a compliance perspective.
```
Comment (mmitoraj) [inline guidelines/releases-guidelines/01-release-strategy.md]:
```suggestion
All Kyma organization members involved in the release process comply with contribution rules defined for Kyma and ensure that all checks built into the continuous integration process pass. These requirements represent the absolute minimal acceptance criteria needed for a release from the compliance perspective.
```

File: guidelines/releases-guidelines/01-release-strategy.md [hunk 6]
```diff
@@ -81,11 +79,11 @@ The Release Manager in Kyma takes care of formal security validation activities
 
 ## Release schedule
 
-A scheduled release follows the planning cycles for the Kyma development that typically take four weeks. At the beginning of each planning cycle, the Head of Capabilities communicates the specific time plan for a release. After reaching the end of the development cycle, Kyma developers create a release candidate which in most cases is identical to the release published on GitHub.
+A scheduled release follows the planning cycles for the Kyma development that typically take four weeks. At the beginning of each planning cycle, the Product Lead communicates the specific time plan for a release. After reaching the end of the development cycle, Kyma developers create a release candidate.
```
Comment (mmitoraj) [inline guidelines/releases-guidelines/01-release-strategy.md]:
```suggestion
A scheduled release follows the planning cycles for the Kyma development that typically take four weeks. At the beginning of each planning cycle, the Product Lead communicates the specific timeframe for a release. After reaching the end of the development cycle, the Release Master creates a release candidate.
```
Kyma developers or the Release Master?

File: guidelines/releases-guidelines/01-release-strategy.md [hunk 7]
```diff
@@ -81,11 +79,11 @@ The Release Manager in Kyma takes care of formal security validation activities
...
 
 All planned releases happen at the end of a planned development cycle. Only the Product Lead and Release Manager can delay or resign from such a release if there are reasons that justify such a decision.
 
-We do not schedule patches but rather provide them on-demand if there is a need to address high priority issues before the next planned release. In this context, we explicitly define high priority issues as those which either affect multiple Kyma installations or result in the financial loss for the users. The Release Manager is involved in the decision to execute a patch release. However, we do not expect to have patch releases before the official Kyma 1.0 version.
+We do not schedule patches but rather provide them on-demand if there is a need to address high priority issues before the next planned release. In this context, we explicitly define high priority issues as those which either affect multiple Kyma installations or result in the financial loss for the users. The Release Manager is involved in the decision to execute a patch release.
```
Comment (mmitoraj) [inline guidelines/releases-guidelines/01-release-strategy.md]:
```suggestion
We do not schedule patches but rather provide them on demand if there is a need to address high priority issues before the next planned release. In this context, we explicitly define high priority issues as those which either affect multiple Kyma installations or result in the financial loss for the users. The Release Manager is involved in the decision to execute a patch release.
```

**Pull request merged**